### PR TITLE
Mock log.Fatalf calls and add test suite

### PR DESCRIPTION
**Pull request by joshua-bone**
/lint

**What this PR does, why we need it**:
- Since log.Fatalf calls are untestable, this PR adds a layer of abstraction and a simple test mock. 
- This PR also adds a complete test suite for the functions in utils.go.
- Increases config-generator test coverage from ~13% to 19.6%.

### CODE REVIEW

File: tools/config-generator/main_test.go [hunk 1]
```diff
@@ -14,17 +14,13 @@ limitations under the License.
 package main
 
 import (
-	"os"
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
 )
 
-func TestMain(m *testing.M) {
```
Comment (coryrc) [inline tools/config-generator/main_test.go:23]: why?

**Comment by coryrc** [comment]
"Since log.Fatalf calls are untestable, this PR adds a layer of abstraction and a simple test mock."

Won't they still cause the test to fail?

File: tools/config-generator/main_test.go (hunk 1 above)
Comment (joshua-bone) [reply tools/config-generator/main_test.go:23]: TestMain does setup only once for the package, not for each individual test as I believed. See this discussion: https://github.com/golang/go/issues/27927

Comment (coryrc) [reply tools/config-generator/main_test.go:23]: oh, ha forgot about that, I thought it was a test called Main... dur

**Comment by joshua-bone** [comment]
> "Since log.Fatalf calls are untestable, this PR adds a layer of abstraction and a simple test mock."
> Won't they still cause the test to fail?

The SetupForTesting() function replaces the logFatalf function with a mock function that simply increments a global counter variable without causing the program to exit.

**Comment by coryrc** [comment]
Correct, but why is this an improvement? In either case, the test correctly
fails.

**Comment by joshua-bone** [comment]
> Correct, but why is this an improvement? In either case, the test correctly
> fails.

There are many cases where I want to verify that an input will cause a failure condition. In those cases I want the code to fail but the test to succeed.

**Comment by coryrc** [comment]
Ah, thanks!

**Comment by coryrc** [comment]
/approve
(I can do that now)

**Pull request merged**